### ISSUE TITLE
Overrides rules provided in `typescript` object is applied before type-awares rules

### ISSUE THREAD

**Issue by TotomInc** [antfu/eslint-config#521]
### Describe the bug

When enabling type-aware rules, the rules provided in the typescript object aren't pushed as the last config items:

```ts
import antfu from '@antfu/eslint-config'

export default antfu({
  typescript: {
    tsconfigPath: './tsconfig.json',

    parserOptions: {
      EXPERIMENTAL_useProjectService: true,
    },

    overrides: {
      // Should not trigger an ESLint error on `main.ts#L4`.
      'ts/no-unsafe-argument': 'off',
    },
  },
})
```

```ts
// main.ts
import { createApp } from 'vue'
import App from './App.vue'

// Unsafe argument of type `any` assigned to a parameter of type `Component<any, any, any, ComputedOptions, MethodOptions, {}, any>`.
// eslint (ts/no-unsafe-argument)
//                    ⌄
const app = createApp(App)
app.mount('#app')
```

![CleanShot 2024-07-01 at 11 15 34](https://github.com/antfu/eslint-config/assets/9285210/8bf3ba1f-f95c-4fed-ac23-c9444680de20)

### Reproduction

https://github.com/TotomInc/antfu-eslint-repro

### System Info

```Shell
System:
    OS: macOS 14.5
    CPU: (8) arm64 Apple M1 Pro
    Memory: 93.69 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.14.0 - ~/.nvm/versions/node/v20.14.0/bin/node
    Yarn: 1.22.19 - ~/.yarn/bin/yarn
    npm: 10.7.0 - ~/.nvm/versions/node/v20.14.0/bin/npm
    pnpm: 8.15.1 - ~/Library/pnpm/pnpm
  Browsers:
    Brave Browser: 126.1.67.123
    Chrome: 126.0.6478.127
    Safari: 17.5
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### Contributions

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests (actually just go ahead and do it, thanks!)

**Comment by Nisgrak** [comment]
I'm having this same issue, maybe a ...overrides in https://github.com/antfu/eslint-config/blob/2e0425685c136f6203adcd5b42474a8d86cfeb36/src/configs/typescript.ts#L54

**Comment by Dimava** [comment]
Workaround: 
```ts
export default antfu({
  typescript: {
    tsconfigPath: './tsconfig.json',

    parserOptions: {
      EXPERIMENTAL_useProjectService: true,
    },
  },
}, {
  files: ['**/*.ts'],
  rules: {
    'ts/no-unsafe-argument': 'off',
  },
})
```
